### DISCUSSION TITLE
notify to every users and keep notification for a long time?

### DISCUSSION THREAD

**Discussion by retsyo** [zauberzeug/nicegui#2016] [Q&A]
### Question

This is an adaption from `nicegui-main\examples\authentication\main.py`. 
In my case, every student users log in, do some exercise and get a score. When he click a button, his name and score should be notified to every users. However my code only notify the user himself. 

So my first question is, any way to fix and notify every user? Thanks
```python
def submit(name, score):
    ui.notify(f'{name} gains {score}', timeout=10000)
    app.storage.user.clear()
    ui.open('/login')

@ui.page('/')
def main_page() -> None:
    with ui.column().classes('absolute-center items-center'):
        name = app.storage.user["username"]

        # following code emulates that student do some exercise and get a score
        random.seed()
        score = random.random()

        ui.label(f'Hello {app.storage.user["username"]}!').classes('text-2xl')
        ui.button(on_click=lambda e: submit(name, score), icon='logout').props('outline round')
```

**Comment by retsyo** [comment]
The 2nd question. I want to keep the notification for a while, as you may read, I used `timeout=10000`. But the notification disappears too quickly yet, no one can read it at all.
Then I added  ` time.sleep(1000)`, but it gives
```
Connection lost.
Trying to reconnect...
``` 
so what is the proper way to keep notification a long life in this case? Thanks

```python
def submit(name, score):
    ui.notify(f'{name} gains {score}', timeout=10000)
    app.storage.user.clear()
    #~ time.sleep(1000)
    ui.open('/login')

@ui.page('/')
def main_page() -> None:
    with ui.column().classes('absolute-center items-center'):
        name = app.storage.user["username"]

        # following code emulates that student do some exercise and get a score
        random.seed()
        score = random.random()

        ui.label(f'Hello {app.storage.user["username"]}!').classes('text-2xl')
        ui.button(on_click=lambda e: submit(name, score), icon='logout').props('outline round')
```

**Reply by falkoschindler** [reply]
If `timeout=10000` is too short, why not increasing it to `timeout=60000` or more?

**Reply by retsyo** [reply]
one more question, no matter which `timeout` is, the notification window for current user disappears quickly, then `/login` html page opens. How to let all user, including current user, have same `timeout` for notify window

**Reply by falkoschindler** [reply]
I can't follow. Have you tried https://github.com/zauberzeug/nicegui/discussions/2016#discussioncomment-7574302?

**Comment by falkoschindler** [comment]
To send a notification to _all_ clients, you can use a custom function like this:
```py
from nicegui import Client

def notify(msg: str):
    for client in Client.instances.values():
        with client:
            ui.notify(msg)
```
